node/impl/full/eth.go

```diff
@@ -570,12 +570,66 @@ func (a *EthModule) EthGetTransactionReceiptLimited(ctx context.Context, txHash
 	return &receipt, nil
 }
 
-func (a *EthAPI) EthGetTransactionByBlockHashAndIndex(context.Context, ethtypes.EthHash, ethtypes.EthUint64) (ethtypes.EthTx, error) {
-	return ethtypes.EthTx{}, ErrUnsupported
+func (a *EthAPI) EthGetTransactionByBlockHashAndIndex(ctx context.Context, blkHash ethtypes.EthHash, index ethtypes.EthUint64) (*ethtypes.EthTx, error) {
+	ts, err := a.Chain.GetTipSetByCid(ctx, blkHash.ToCid())
+	if err != nil {
+		return nil, xerrors.Errorf("block not found")
+	}
+
+	return a.getTransactionByTipsetAndIndex(ctx, ts, index)
+}
+
+func (a *EthAPI) EthGetTransactionByBlockNumberAndIndex(ctx context.Context, blkParam string, index ethtypes.EthUint64) (*ethtypes.EthTx, error) {
+
+	ts, err := getTipsetByBlockNumber(ctx, a.Chain, blkParam, true)
+	if err != nil {
+		if err == ErrNullRound {
+			return nil, nil
+		}
```

Comment:
See https://github.com/filecoin-project/lotus/issues/12633 that I just wrote up today, let's not introduce another case of `null` for null rounds here since only 1 of our current APIs do this and apparently (TIL) that's not in line with what an Eth dev would expect `null` to mean, we should just return the right error here.

Reply:
I'll add `return nil, xerrors.Errorf("block number %s was a null round", blkParam)` and same for `EthGetTransactionByBlockHashAndIndex` as well